### PR TITLE
cds extractor: npm install in each `package.json`

### PR DESCRIPTION
**Pull request by lcartey**
This is because cds dependencies can exist in the package.json that may need to be installed before running the cds compiler.

### CODE REVIEW

File: extractors/cds/tools/index-files.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ then
     # directory.
     #
     # We also ensure we skip node_modules, as we can end up in a recursive loop
-    find . -type d -name node_modules -prune -false -o -type f \( -iname 'package.json' \) -exec grep -ql '@sap/cds' {} \; -execdir bash -c "grep -q \"^\$(pwd)\(/\|$\)\" \"$response_file\"" \; -execdir bash -c "echo \"Installing @sap/cds-dk into \$(pwd) to enable CDS compilation.\"" \; -execdir npm install --silent @sap/cds-dk \;
+    find . -type d -name node_modules -prune -false -o -type f \( -iname 'package.json' \) -exec grep -ql '@sap/cds' {} \; -execdir bash -c "grep -q \"^\$(pwd)\(/\|$\)\" \"$response_file\"" \; -execdir bash -c "echo \"Installing @sap/cds-dk into \$(pwd) to enable CDS compilation.\"" \; -execdir npm install --silent @sap/cds-dk \; -execdir npm install --silent \;
```
Comment (jeongsoolee09) [inline extractors/cds/tools/index-files.sh:39]: A quick thought: would it make sense to install `@sap/cds-dk` globally? Then we don't have to run this `find` command.

Comment (lcartey) [reply extractors/cds/tools/index-files.sh:39]: Unfortunately not. We have seen cases of monorepos where different versions of `@sap/cds` are specified in different directories. The validity of the CDS file format can change between major versions, so it's not always possible to just install one version and have it work everywhere. This approach ensures we install a compatible version of the `@sap/cds-dk` tools for each directory, minimising the chance of failure.

Comment (jeongsoolee09) [reply extractors/cds/tools/index-files.sh:39]: Then we'll have to do it project by project then. Thanks!

**Review by jeongsoolee09 — APPROVED** [review]
Thank you!

**Pull request merged**